### PR TITLE
[gamecontroller] Add nullability to (generated and manual) bindings

### PR DESCRIPTION
**Pull request by tj-devel709**
This PR aims to bring nullability changes to GameController.
Following the steps [here](https://github.com/xamarin/xamarin-macios/wiki/Nullability):

1. I am adding `nullable enable` to all manual files that are not "API_SOURCES" in src/frameworks.sources and making the required nullability changes
2. Changing any `== null` or `!= null` to `is null` and `is not null`

### CODE REVIEW

**Review by stephen-hawley — COMMENTED** [review]
Later on in the GCMotion.cs - is this valid?
```
public GCDualSenseAdaptiveTriggerPositionalAmplitudes (float [] values)
		{
			if (values is null)
				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (values));
```

if I write this:
```
		public void Foo (float [] values)
		{
			if (values is null)
				throw new ArgumentNullException (nameof (values));
		}

		public void UseFoo ()
		{
			Foo (null);
		}
```
In trying to call Foo, I get "Cannot convert null literal to non-null reference type"

**Review by mandel-macaque — APPROVED** [review]

**Comment by tj-devel709** [comment]
> Later on in the GCMotion.cs - is this valid?
> 
> ```
> public GCDualSenseAdaptiveTriggerPositionalAmplitudes (float [] values)
> 		{
> 			if (values is null)
> 				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (values));
> ```
> 
> if I write this:
> 
> ```
> 		public void Foo (float [] values)
> 		{
> 			if (values is null)
> 				throw new ArgumentNullException (nameof (values));
> 		}
> 
> 		public void UseFoo ()
> 		{
> 			Foo (null);
> 		}
> ```
> 
> In trying to call Foo, I get "Cannot convert null literal to non-null reference type"

@stephen-hawley  hmmm interesting point, I think maybe this would fall under what @mandel-macaque  said here https://github.com/xamarin/xamarin-macios/pull/14611#discussion_r840913912 that if someone doesn't have nullability enabled on their project, they won't see the compiler error

**Review by dalexsoto — APPROVED** [review]

File: src/GameController/GCExtendedGamepadSnapshot.cs [hunk 1]
```diff
@@ -325,7 +327,7 @@ public static bool TryGetSnapShotData (NSData data, out GCExtendedGamepadSnapSho
 #endif
 		public static bool TryGetExtendedSnapShotData (NSData data, out GCExtendedGamepadSnapshotData snapshotData)
```
Comment (rolfbjarne) [inline src/GameController/GCExtendedGamepadSnapshot.cs:328]: Same:

```suggestion
		public static bool TryGetExtendedSnapShotData (NSData? data, out GCExtendedGamepadSnapshotData snapshotData)
```

File: src/GameController/GCExtendedGamepadSnapshot.cs [hunk 2]
```diff
@@ -301,7 +303,7 @@ public partial class GCExtendedGamepadSnapshot {
 
 		public static bool TryGetSnapShotData (NSData data, out GCExtendedGamepadSnapShotDataV100 snapshotData)
```
Comment (rolfbjarne) [inline src/GameController/GCExtendedGamepadSnapshot.cs:304]: If the parameter can be null, you should also change it to say as much:

```suggestion
		public static bool TryGetSnapShotData (NSData? data, out GCExtendedGamepadSnapShotDataV100 snapshotData)
```

File: src/GameController/GCGamepadSnapshot.cs [hunk 3]
```diff
@@ -77,7 +79,7 @@ public partial class GCGamepadSnapshot {
 
 		public static bool TryGetSnapshotData (NSData data, out GCGamepadSnapShotDataV100 snapshotData)
```
Comment (rolfbjarne) [inline src/GameController/GCGamepadSnapshot.cs:80]:
```suggestion
		public static bool TryGetSnapshotData (NSData? data, out GCGamepadSnapShotDataV100 snapshotData)
```

File: src/GameController/GCMicroGamepadSnapshot.cs [hunk 4]
```diff
@@ -206,7 +208,7 @@ public partial class GCMicroGamepadSnapshot {
 #endif
 		public static bool TryGetSnapshotData (NSData data, out GCMicroGamepadSnapShotDataV100 snapshotData)
```
Comment (rolfbjarne) [inline src/GameController/GCMicroGamepadSnapshot.cs:209]:
```suggestion
		public static bool TryGetSnapshotData (NSData? data, out GCMicroGamepadSnapShotDataV100 snapshotData)
```

File: src/GameController/GCMicroGamepadSnapshot.cs [hunk 5]
```diff
@@ -256,7 +258,7 @@ public static bool TryGetSnapshotData (NSData data, out GCMicroGamepadSnapShotDa
 #endif
 		public static bool TryGetSnapshotData (NSData data, out GCMicroGamepadSnapshotData snapshotData)
```
Comment (rolfbjarne) [inline src/GameController/GCMicroGamepadSnapshot.cs:259]:
```suggestion
		public static bool TryGetSnapshotData (NSData? data, out GCMicroGamepadSnapshotData snapshotData)
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Comment by tj-devel709** [comment]
@rolfbjarne this one is ready for when you return :)

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**